ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -123,6 +126,7 @@ public VcrReplicationManager(VerifiableProperties properties, StoreManager store
     this.azureCloudConfig = new AzureCloudConfig(properties);
     this.vcrMetrics = new VcrMetrics(metricRegistry);
     this.azureMetrics = new AzureMetrics(metricRegistry);
+    this.azureStorageContainerMetricsCollector = AzureStorageContainerMetricsCollector.getInstance(metricRegistry, properties);
```

Comment:
if we are passing it to every VcrReplicaThread, then why have a static instance method? 
You can just have a regular constructor for this collector class.

Reply:
Good pt ! Changed this such that each thread gets the instance without passing it to every thread ctor.